src/object/srv_obj.c

```diff
@@ -3206,16 +3189,12 @@ obj_local_enum(struct obj_io_context *ioc, crt_rpc_t *rpc,
 		goto failed;
 
 re_pack:
-	rc = ds_obj_enum_pack(&param, type, recursive, &anchors[0], enum_arg, vos_iterate, dth);
+	rc = ds_obj_enum_pack(&param, type, recursive, anchors, enum_arg, vos_iterate, dth);
 	if (obj_dtx_need_refresh(dth, rc)) {
 		rc = dtx_refresh(dth, ioc->ioc_coc);
-		if (rc == -DER_AGAIN) {
-			anchors[0] = anchors[1];
-			obj_restore_enum_args(rpc, enum_arg, &saved_arg);
-			if (opc == DAOS_OBJ_RPC_ENUMERATE)
-				fill_oid(oei->oei_oid, enum_arg);
```

Comment:
could you please explain a little bit why above restore section is not needed any more? thx

Reply:
Because vos_iterate() takes **anchor** parameter, that is both input and output. When vos_iterate() returns for retry, it contains enough information to locate the position for next item. When retry after DTX refresh, it is better to resume from break point (that is the one triggered DTX refresh) instead of from the relative beginning with restore() method.